### PR TITLE
[DSM] Update appengine app.yaml for Dev env

### PR DESCRIPTION
**Pull request by aweng98**
## Context

[Google Cloud Automatica Scaling](https://cloud.google.com/appengine/docs/legacy/standard/python/config/appref#scaling_elements)

On Dev only, updated appengine `app.yaml` to handle slower request time from DSM back-end server when server is under load from Playwright tests.

**Issue**:
Appengine automatically aborts some DSM application requests when request pending time has exceeded max pending time.
Error signature: `Request was aborted after waiting too long to attempt to service your request.`

Screenshot 2023-10-13 at 12 01 25 PM

**Cause**:
Running Playwright tests concurrently on several CircleCI VMs causes DSM back-end server response time increases significantly.

Screenshot of GCP logs that confirms this error happens a lot ( 270 times in the last 30 days on Dev):

Screenshot 2023-10-13 at 11 58 24 AM

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have added regression test labels to the jira ticket
- [ ] I have added verification steps to the jira ticket
- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/config/DataDonationPlatform.yaml.ctmpl [hunk 1]
```diff
@@ -13,8 +13,13 @@ manual_scaling:
 manual_scaling:
   instances: 2
 {{else if eq $environment "dev"}}
-manual_scaling:
-  instances: 2
+automatic_scaling:
+  max_instances: 2
+  min_instances: 1
+  target_cpu_utilization: 0.95
```
Comment (aweng98) [inline pepper-apis/config/DataDonationPlatform.yaml.ctmpl:19]: `target_cpu_utilization`: default value is 0.6.

File: pepper-apis/config/DataDonationPlatform.yaml.ctmpl [hunk 2]
```diff
@@ -13,8 +13,13 @@ manual_scaling:
...
+  target_throughput_utilization: 0.95
```
Comment (aweng98) [inline pepper-apis/config/DataDonationPlatform.yaml.ctmpl:20]: `target_throughput_utilization`: default value is 0.6.

File: pepper-apis/config/DataDonationPlatform.yaml.ctmpl [hunk 3]
```diff
@@ -13,8 +13,13 @@ manual_scaling:
...
+  max_concurrent_requests: 80
```
Comment (aweng98) [inline pepper-apis/config/DataDonationPlatform.yaml.ctmpl:21]: `max_concurrent_requests`: default value is 10.

File: pepper-apis/config/DataDonationPlatform.yaml.ctmpl [hunk 4]
```diff
@@ -13,8 +13,13 @@ manual_scaling:
...
+  max_pending_latency: 30s
```
Comment (aweng98) [inline pepper-apis/config/DataDonationPlatform.yaml.ctmpl:22]: `max_pending_latency`: default is 10 seconds.

**Review by Rinchan20 — APPROVED** [review]

**Review by zyme — CHANGES_REQUESTED** [review]
I have concerns that this will lead to unexpected behavior across dev/test/staging as well as prod.  We should have the same infrastructure settings across all environments.  Our prior experience tuning automatic scaling in GAE did not turn out well.  Let's discuss whether we want to give this another try.

**Pull request closed without merging**